crates/typst-pdf/src/font.rs

```diff
@@ -23,6 +26,110 @@ const SYSTEM_INFO: SystemInfo = SystemInfo {
 /// Embed all used fonts into the PDF.
 #[typst_macros::time(name = "write fonts")]
 pub(crate) fn write_fonts(ctx: &mut PdfContext) {
+    let color_font_map = ctx.color_font_map.take_map();
+    for (font_info, font) in color_font_map {
+        for (font_index, subfont_id) in font.refs.iter().enumerate() {
```

Comment:
This part could use some more comments, at least one per "paragraph" (i.e. block of consecutive lines of code).